heudiconv/convert.py

```diff
@@ -975,9 +979,17 @@ def save_converted_files(
         is_uncombined = (
             len(set(filter(bool, channel_names))) > 1
         )  # Check for uncombined data
-        is_complex = (
-            "M" in image_types and "P" in image_types
-        )  # Determine if data are complex (magnitude + phase)
+        CPLX_PARTS = ["MAGNITUDE", "PHASE", "IMAGINARY", "REAL"]
+        is_complex = len(
+            set(
+                [
+                    part
+                    for its in image_types
+                    for part in CPLX_PARTS
+                    if part in its or part[0] in its
+                ]
+            )
+        )  # Determine if data are complex (magnitude + phase or real + imag or all-4)
```

Comment:
```suggestion
        is_complex = len(
        set(
            its
            for its in image_types
            if any(part == its for part in CPLX_PARTS)
        )
        ) # Determine if data are complex (magnitude + phase or real + imag or all-4)
```
Otherwise there will be some image_types (e.g., "IMAGINARY") for which the first letter "I" is in "PRIMARY" or in "ORIGINAL" resulting in is_complex = True. Trowing errors when the update_complex_name function is run.